src/components/BookmarkCard/Container.tsx

```diff
@@ -0,0 +1,52 @@
+import classNames from 'classnames';
+import type { PropsWithChildren } from 'react';
+import React, { useRef, useState } from 'react';
+import { useResizeObserver } from '@react-hookz/web';
+import { StoryBookmarkLayout } from '@prezly/slate-types';
+
+type ContainerLayout = 'vertical' | 'horizontal';
+
+interface ContainerProps {
+    defaultLayout: ContainerLayout;
+    hasThumbnail: boolean;
+    className?: string;
+}
+
+const HORIZONTAL_LAYOUT_MIN_WIDTH = 480;
+
+export function Container({
+    defaultLayout,
+    children,
+    className,
+    hasThumbnail,
+}: PropsWithChildren<ContainerProps>) {
+    const card = useRef<HTMLDivElement | null>(null);
+    const [isSmallViewport, setSmallViewport] = useState(false);
+
+    useResizeObserver(card.current, function (entry) {
+        setSmallViewport(entry.contentRect.width < HORIZONTAL_LAYOUT_MIN_WIDTH);
+    });
```

Comment:
This doesn't seem to work. Neither in Firefox, nor in Chromium.

The card should switch to vertical layout, but it doesn't:

![Screenshot from 2022-04-19 13-07-25](https://user-images.githubusercontent.com/370680/163981086-22a91606-9292-48e9-a01a-c91bf65d76c1.png)